### PR TITLE
Implement RateCommand v1.1

### PR DESCRIPTION
**Pull request by IamRENCE**
Currently, PersonCard only stores information like Name, Email, Address, Phone and Tags. There is an absence of a field where the user can add/edit a person's (or rather employee's) rating.

A 'rate' command should therefore be implemented so as to allow managers to review their staff.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/RateCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RateCommand.MESSAGE_USAGE));
+        }
+        rating = argMultimap.getValue(PREFIX_RATING).get();
+        index = ParserUtil.parseIndex(argMultimap.getPreamble());
+
+        return new RateCommand(index, new Rating(rating));
+    }
+
+    /**
+     * Returns true if prefix contains empty {@code Optional} values in the given
```
Comment (DanKhoo) [inline src/main/java/seedu/address/logic/parser/RateCommandParser.java]: Returns true if no prefix contains empty values 
or 
Returns true if prefix contains empty values

File: src/main/java/seedu/address/model/person/Rating.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Represents a Person's rating number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRating(String)}
+ */
+public class Rating {
+
+    public static final Rating DEFAULT_INITIAL_RATING = new Rating("0");
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ratings should only contain numbers, and it should be between 0 to 10";
```
Comment (DanKhoo) [inline src/main/java/seedu/address/model/person/Rating.java]: 1 to 10?

File: src/main/java/seedu/address/model/person/Rating.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public static final String VALIDATION_REGEX = "[0-9]|10";
```
Comment (DanKhoo) [inline src/main/java/seedu/address/model/person/Rating.java]: Validate for 1-10

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -37,6 +37,8 @@
     @FXML
     private Label email;
     @FXML
+    private Label rating;
```
Comment (DanKhoo) [inline src/main/java/seedu/address/ui/PersonCard.java]: Rating not needed in PersonCard already

File: src/main/resources/view/PersonListCard.fxml [hunk 5]
```diff
@@ -31,6 +31,7 @@
       <Label fx:id="phone" styleClass="cell_small_label" text="\$phone" />
       <Label fx:id="address" styleClass="cell_small_label" text="\$address" />
       <Label fx:id="email" styleClass="cell_small_label" text="\$email" />
+      <Label fx:id="rating" styleClass="cell_small_label" text="\$rating" />
```
Comment (DanKhoo) [inline src/main/resources/view/PersonListCard.fxml]: Rating not needed in the personListCard already

File: src/test/java/guitests/guihandles/PersonCardHandle.java [hunk 6]
```diff
@@ -19,13 +19,15 @@
     private static final String ADDRESS_FIELD_ID = "#address";
     private static final String PHONE_FIELD_ID = "#phone";
     private static final String EMAIL_FIELD_ID = "#email";
+    private static final String RATING_FIELD_ID = "#rating";
```
Comment (DanKhoo) [inline src/test/java/guitests/guihandles/PersonCardHandle.java]: Ratings not needed in PersonCardHandle.java as well

File: src/main/java/seedu/address/logic/parser/RateCommandParser.java (hunk 1 above)
Comment (IamRENCE) [reply src/main/java/seedu/address/logic/parser/RateCommandParser.java]: Should be if no prefix, thanks for highlighting

**Pull request merged**